packages/paste-core/components/menu/src/MenuBadge.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {Box} from '@twilio-paste/box';
+import {Badge} from '@twilio-paste/badge';
+import type {BadgeProps} from '@twilio-paste/badge';
+import {Button} from '@twilio-paste/button';
+import type {ButtonProps} from '@twilio-paste/button';
+import {MenuPrimitiveButton} from '@twilio-paste/menu-primitive';
+import type {MenuPrimitiveButtonProps} from '@twilio-paste/menu-primitive';
+import {ChevronDownIcon} from '@twilio-paste/icons/esm/ChevronDownIcon';
+
+export type MenuBadgeProps = MenuPrimitiveButtonProps &
+  Omit<ButtonProps, 'variant'> &
+  Pick<BadgeProps, 'variant'> & {
+    buttonLabel: string;
+  };
+
+export type MenuBadgeContentProps = Omit<MenuBadgeProps, 'variant'>;
+export const MenuBadgeContent = React.forwardRef<HTMLButtonElement, MenuBadgeContentProps>(
+  ({children, element, buttonLabel, ...props}, ref) => {
+    return (
+      <Box as="span" display="flex" flexDirection="row" columnGap="space20" alignItems="center">
+        {children}
+        <MenuPrimitiveButton
+          {...props}
+          variant="secondary_icon"
+          size="reset"
+          element={`${element}_BUTTON`}
+          as={Button}
+          ref={ref}
+        >
+          <ChevronDownIcon decorative={false} size="sizeIcon10" title={buttonLabel} />
+        </MenuPrimitiveButton>
+      </Box>
+    );
+  }
+);
+MenuBadgeContent.displayName = 'MenuBadgeContent';
+
+const MenuBadge = React.forwardRef<HTMLButtonElement, MenuBadgeProps>(
+  ({children, element = 'MENU_BADGE', variant = 'default', buttonLabel, ...props}, ref) => {
+    return (
+      <Badge element={element} as="span" variant={variant}>
+        <MenuBadgeContent {...props} element={element} buttonLabel={buttonLabel} ref={ref}>
```

Comment:
suggestion: prefix the label with i18n, so `i18nButtonLabel`. So it matches what we do in Modals for the Close button and works out of the box with Consoles i18n linter (I think)

Reply:
Probably best to follow the rest of the system